File: scripts/download_monument_images.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""Télécharge une image placeholder pour chaque monument sans image dans assets/monuments/."""
import re
import os
import ssl
import urllib.request
import urllib.error

DATA_FILE = os.path.join(os.path.dirname(__file__), '..', 'monuments-data.js')
OUT_DIR = os.path.join(os.path.dirname(__file__), '..', 'assets', 'monuments')

def extract_monuments(content):
    list = []
    for m in re.finditer(r"\{\s*id:\s*'([^']+)',\s*name:\s*'((?:[^'\\]|\\.)*)'", content):
        name = m.group(2).replace("\\'", "'")
        list.append({'id': m.group(1), 'name': name})
    return list

def main():
    with open(DATA_FILE, 'r', encoding='utf-8') as f:
        monuments = extract_monuments(f.read())
    os.makedirs(OUT_DIR, exist_ok=True)
    downloaded = skipped = 0
    for m in monuments:
        out_path = os.path.join(OUT_DIR, m['id'] + '.png')
        if os.path.exists(out_path):
            skipped += 1
            continue
        text = urllib.parse.quote((m['name'] or '')[:20])
        url = 'https://placehold.co/200x120/1a1a2e/f0c14b?text=' + text
        try:
            ctx = ssl.create_default_context()
            ctx.check_hostname = False
            ctx.verify_mode = ssl.CERT_NONE
            req = urllib.request.Request(url, headers={'User-Agent': 'Python-download'})
            with urllib.request.urlopen(req, timeout=15, context=ctx) as r:
                with open(out_path, 'wb') as f:
                    f.write(r.read())
            downloaded += 1
            print('OK', m['id'])
        except Exception as e:
            print('Erreur', m['id'], str(e))
    print('Terminé:', downloaded, 'téléchargées,', skipped, 'déjà présentes.')

if __name__ == '__main__':
    main()

File: scripts/download-monument-images.js
/**
 * Télécharge une image placeholder (avec le nom du monument) pour chaque monument
 * qui n'a pas encore d'image dans assets/monuments/.
 * Usage: node scripts/download-monument-images.js
 */
const https = require('https');
const fs = require('fs');
const path = require('path');

const DATA_FILE = path.join(__dirname, '..', 'monuments-data.js');
const OUT_DIR = path.join(__dirname, '..', 'assets', 'monuments');

function fetchUrl(url) {
  return new Promise((resolve, reject) => {
    https.get(url, { headers: { 'User-Agent': 'Node-download' } }, (res) => {
      const chunks = [];
      res.on('data', (chunk) => chunks.push(chunk));
      res.on('end', () => resolve(Buffer.concat(chunks)));
      res.on('error', reject);
    }).on('error', reject);
  });
}

function extractMonuments(content) {
  const list = [];
  const re = /\{\s*id:\s*'([^']+)',\s*name:\s*'((?:[^'\\]|\\.)*)'/g;
  let m;
  while ((m = re.exec(content)) !== null) {
    const name = m[2].replace(/\\'/g, "'");
    list.push({ id: m[1], name });
  }
  return list;
}

async function main() {
  const raw = fs.readFileSync(DATA_FILE, 'utf8');
  const monuments = extractMonuments(raw);
  if (!fs.existsSync(OUT_DIR)) fs.mkdirSync(OUT_DIR, { recursive: true });

  let downloaded = 0;
  let skipped = 0;
  for (const { id, name } of monuments) {
    const outPath = path.join(OUT_DIR, id + '.png');
    if (fs.existsSync(outPath)) {
      skipped++;
      continue;
    }
    const text = encodeURIComponent(name.length > 20 ? name.slice(0, 20) : name);
    const url = `https://placehold.co/200x120/1a1a2e/f0c14b?text=${text}`;
    try {
      const buf = await fetchUrl(url);
      fs.writeFileSync(outPath, buf);
      downloaded++;
      console.log('OK', id);
    } catch (e) {
      console.error('Erreur', id, e.message);
    }
  }
  console.log('Terminé:', downloaded, 'téléchargées,', skipped, 'déjà présentes.');
}

main().catch(console.error);
